src/code/z_player_lib.c

```diff
@@ -1,153 +1,1021 @@
+    f32 phi_f12;
+
+    if (*arg0 < arg2) {
+        *arg1 += arg3;
+    } else {
+        *arg1 -= arg3;
+    }
+
+    *arg1 = CLAMP(*arg1, -arg4, arg4);
+    *arg0 += (s16)*arg1;
+    if (((arg2 - *arg0) * (s16)*arg1) < 0) {
+        *arg0 = arg2;
+    }
+}
+
+void func_80124FF0(f32 arg0, s16 arg1, Vec3f* arg2, s16 arg3, Vec3f* arg4, Vec3f* arg5, s16* arg6, f32* arg7, f32 arg8,
+                   s16 arg9, s16* arg10, f32* arg11, s32 arg12) {
+    Vec3f sp44;
+    f32 sp40;
+    s16 temp_v0;
+    s16 sp3C;
+    f32 temp_f14;
+    f32 sp34 = Math_CosS(arg1) * arg0;
+    f32 sp30 = Math_SinS(arg1) * -arg0;
+
+    arg5->x = (Math_SinS(arg3) * sp30) + arg2->x;
+    arg5->y = arg2->y + sp34;
+    arg5->z = (Math_CosS(arg3) * sp30) + arg2->z;
+
+    Math_Vec3f_Diff(arg5, arg4, &sp44);
+    sp40 = sqrtf(SQ(sp44.x) + SQ(sp44.z));
+
+    sp3C = (sp40 <= 1.0f) ? arg3 : Math_FAtan2F(sp44.z, sp44.x);
+    sp40 = (Math_CosS(sp3C - arg3) * sp40) + arg8;
+
+    if (ABS_ALT(BINANG_SUB(sp3C, arg3)) > 0x4000) {
+        sp3C = BINANG_ROT180(sp3C);
+    }
+    sp3C = sp3C - arg3;
+
+    temp_v0 = Math_FAtan2F(sp44.y, sp40);
+    temp_v0 = CLAMP(temp_v0, (s16)-arg9, arg9);
+    // fake match?
+    if (sp3C) {}
+
+    func_80124F18(arg6, arg7, temp_v0, 20.0f, 2000.0f);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801241E0.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801251C4.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_8012420C.s")
+void func_80125318(Vec3f* arg0, Vec3s* arg1) {
+    arg0->x = 0.0f;
+    arg0->y = 0.0f;
+    arg0->z = 0.0f;
+    arg1->x = 0;
+    arg1->y = 0;
+    arg1->z = 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_8012422C.s")
+void func_80125340(void) {
+    Matrix_StatePush();
+    D_801C0958 = true;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/Player_GetExplosiveHeld.s")
+void func_8012536C(void) {
+    if (D_801C0958) {
+        Matrix_StatePop();
+        D_801C0958 = false;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80124278.s")
+// draws zora shield (?)
+void func_801253A4(GlobalContext* globalCtx, Player* player) {
+    u8* phi_a0;
+    Vtx* sp30;
+    Gfx* dl;
+    f32 sp28; // scale
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801242B4.s")
+    sp28 = player->unk_B62 * 0.19607843f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801242DC.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80124420.s")
+    AnimatedMat_DrawXlu(globalCtx, Lib_SegmentedToVirtual(&object_link_zora_Matanimheader_012A80));
+    Matrix_Scale(sp28, sp28, sp28, 1);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80124618.s")
+    // clang-format off
+    sp30 = Lib_SegmentedToVirtual(&object_link_zora_Vtx_011210), phi_a0 = Lib_SegmentedToVirtual(&object_link_zora_U8_011710);
+    // clang-format on
 
```

Comment:
Is this needed even with a `,` instead of a `;`?

Reply:
No, I'll change it